oak-core/src/main/java/org/apache/jackrabbit/oak/Oak.java

```diff
@@ -576,6 +577,11 @@ public Oak with(@NotNull Whiteboard whiteboard) {
             newSettings.setLimitReads(this.queryEngineSettings.getLimitReads());
             this.queryEngineSettings = new AnnotatedQueryEngineSettings(newSettings);
         }
+        if (queryEngineSettings != null) {
```

Comment:
Could this be a mistake? We have an environment where the FT gets never loaded and I think it could be because of this check. It seems there is a local variable that was initialized above:

`QueryEngineSettings queryEngineSettings = WhiteboardUtils.getService(whiteboard, QueryEngineSettings.class);`

But probably this condition should check if the field is null, instead of that local variable:

`if (this.queryEngineSettings != null) {`

Could somebody confirm this? And, in case this is expected and I'm wrong, could you help me to know what config would be needed to enable the FT in an environment?

/cc @thomasmueller @nit0906 @joerghoh @mreutegg @stefan-egli 
PD: Sorry for pinging directly, but as the PR is already closed I'm not sure if notifications will arrive otherwise.